### ISSUE TITLE
Is there a way to enable forced stereo output in the SDP?

### ISSUE THREAD

**Issue by ZenithMDC** [Glimesh/broadcast-box#33]
Chrome downmixes the audio to mono by default. I had assumed it would be as simple as modifying the string on this line: https://github.com/Glimesh/broadcast-box/blob/51c0655751995ab790bd6778ccc4a2a855f49916/internal/webrtc/webrtc.go#L158
And replacing it with `minptime=10;useinbandfec=1;stereo=1`, but it seems to have no effect. I tried a few other things, too, but to no avail.

So, I'm not at all sure how this is working. It seems like the SDP is getting filled with `minptime=10;useinbandfec=1` somewhere else.

If you have any ideas about this, please let me know. Thanks in advance.

*EDIT:*
I managed to get this to work by modifying `index.js` in `web/src/components/player/`:
```diff
diff --git a/web/src/components/player/index.js b/web/src/components/player/index.js
index 21f611f..5d9911e 100644
--- a/web/src/components/player/index.js
+++ b/web/src/components/player/index.js
@@ -66,6 +66,7 @@ function Player({ cinemaMode }) {
     peerConnection.addTransceiver('video', { direction: 'recvonly' })

     peerConnection.createOffer().then(offer => {
+      offer["sdp"] = offer["sdp"].replace("useinbandfec=1", "useinbandfec=1;stereo=1")
       peerConnection.setLocalDescription(offer)

       fetch(`${process.env.REACT_APP_API_PATH}/whep`, {
```
I'll leave this issue open since it seems more like a hack than anything else and perhaps there's a more elegant solution available, but this was the only solution I've tried so far that worked.

**Comment by withorwithoutvoices** [comment]
Dear Senith,

SDP console programmation complicate so much your life?

You must bike vigorously to Lanus Grabacion department of audiovisual and start career, or you will loss quality. By that, you will avoid persecution of Youtube copywrite (sir).

TLDR; they not support loud noises, so you don't have a choice. 

Thanks -- for your support. I end the message. 

- Flavio "Different Horses for Different Courses" Ontivero

**Comment by mpisat** [comment]
A similar hack is used by other players such as OvenPlayer and my custom Wowza player.
You should send a MR for broadcast-box FE code.

**Comment by Sean-Der** [comment]
Hey @ZenithMDC 

Mind opening a PR. I am totally ok with landing that.

Doesn't look like any good Javascript APIs exist for this today :( just want to give people the best experience possible.

**Comment by ZenithMDC** [comment]
Oops... Sorry, I don't check GitHub or my email as often as I should. I'll get right to it...
